Prog_3.cpp

```diff
@@ -0,0 +1,73 @@
+﻿#include <iostream>
+#include <cmath>
+using namespace std;
+
+class Triangle {
+private:
+    double a;
+    double b;
+    double c;
+
+public:
+    Triangle() {
+        a = 1;
+        b = 1;
+        c = 1;
+    }
+
+    Triangle(double a_, double b_, double c_) {
+        a = a_;
+        b = b_;
+        c = c_;
+    }
+
+    double getA() { return a; }
+    void setA(double a_) { a = a_; }
+
+    double getB() { return b; }
+    void setB(double b_) { b = b_; }
+
+    double getC() { return c; }
+    void setC(double c_) { c = c_; }
+
+    int La() { return acos((a * a + c * c - b * b) / (2 * a * c)) * 180.0 / 3.14; }
+    int Lb() { return acos((a * a + b * b - c * c) / (2 * a * b)) * 180.0 / 3.14; }
+    int Lc() { return acos((b * b + c * c - a * a) / (2 * b * c)) * 180.0 / 3.14; }
+
+    double P() { return a + b + c; }
+};
+
+class Rightangled : public Triangle {
+private:
+    double S;
+public:
+    Rightangled() : Triangle() {
+        setS();
+    }
+
+    Rightangled(double a_, double b_, double c_) :Triangle(a_, b_, c_) {
+        setS();
+    }
+
+    void setS() {
+        if (La() == 90) S = 0.5 * getB() * getC();
+        else if (Lb() == 90) S = 0.5 * getA() * getC();
+        else if (Lc() == 90) S = 0.5 * getA() * getB();
+        else cout << "Ошибка! Треугольник не прямоугольный!" << endl;
+    }
+
+    double getS() { return S; }

```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.